ITISFoundation/osparc-simcore
# 🐛 Fix bug when using non-debian `hostname` command in Makefiles

## What do these changes do?

Simcore implicitly assumed that people run a debian-distribution based linux, since in the `get_my_ip` Makefile functions CLI options exclusive to the debian-brand `hostname` command are used. The code of this debian `hostname` implementation can only found inside debian, and it is not a common binary outside debain-linux-distributions. See: https://github.com/coreos/bugs/issues/1983#issuecomment-304911238 . The most canonical `hostname` executable would likely be the GNU/inetutils implementation, which is available on basically all distros as part of `coreutils`. 

I have added a bash-style OR-statement (`||`) that will fallback to a GNU/inetutils-style call to `hostname` if the previous call with debian syntax (`hostname --all-ip-addresses`) exits with an error-code.

**Bonus:**
- Fix typos
- Make flaky test less flaky: `tests/unit/with_dbs/01/studies_dispatcher/test_studies_dispatcher_studies_access.py:212` can fail locally if the machine is not fast enough, with `AssertionError: Expected /study/uuid, got /error?message=We+have+reached+the+maximum+of+anonymous+users+allowed+the+platform.+Please+try+later+or+login+with+a+registered+account.&status_code=429`. A retry-clause fixes this local flakyness.

## How to test

Run `make install-dev` somewhere, observe no errors if you are running a non-debian `hostname` command.
## Dev-ops checklist
